core/src/weighted_shuffle.rs

```diff
@@ -21,8 +21,8 @@ where
             (
                 i,
                 (&mut rng)
-                    .gen_range(1, std::u16::MAX as u32)
-                    .saturating_mul(x),
+                    .gen_range(1, u32::from(std::u16::MAX))
+                    .saturating_mul(x) as u64,
```

Comment:
We should remove the saturated mul. Capture the generated range into a u64 and then just multiply. Right now the as u64 doesn't do anything because the mul could have already saturated.

Reply:
this? ```(&mut rng).gen_range(1, u64::from(std::u16::MAX)) * u64::from(x)```

Reply:
Yeah i think that works.

Reply:
👍